x-pack/plugins/cases/server/routes/api/comments/get_all_comment.test.ts

```diff
@@ -6,16 +6,30 @@
  */
 
 import { getAllCommentsRoute } from './get_all_comment';
+import { docLinksServiceMock } from '@kbn/core/server/mocks';
 
 describe('getAllCommentsRoute', () => {
+  const docLinks = docLinksServiceMock.createSetupContract();
+
   it('marks the endpoint internal in serverless', async () => {
-    const router = getAllCommentsRoute({ isServerless: true });
+    const router = getAllCommentsRoute({ isServerless: true, docLinks });
 
     expect(router.routerOptions?.access).toBe('internal');
+    expect(router.routerOptions?.deprecated).toMatchInlineSnapshot(`
```

Comment:
nit: I think it would be better if we had a dedicated test for the deprecation object check. The object is available in all flavors, serverless or non-serverless. Same for all test files.